### PR TITLE
feat: 736 Feature Flag

### PR DESCRIPTION
**Pull request by k8hansen**

### CODE REVIEW

File: src/pages/_app.tsx [hunk 1]
```diff
@@ -34,7 +36,7 @@ const CustomApp: FC<AppProps> = ({ Component, pageProps }) => {
   // }, [router.events]);
 
   return (
-    <>
+    <SSRProvider>
```
Comment (k8hansen) [inline src/pages/_app.tsx:39]: Do we still need this?

File: src/hooks/useFeatureToggle.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+import { FeatureToggleContext } from '../context/featureToggleContext';
+
+const useFeatureToggle = () => {
+  const { enabledFeatures } = useContext(FeatureToggleContext);
+  const isEnabled = (featureName: string) =>
+    enabledFeatures.includes(featureName);
+
+  return [isEnabled];
+};
```
Comment (mewdriller) [inline src/hooks/useFeatureToggle.ts:11]: 💭 Maybe a shift in the shape of this to something like:

```tsx
import { useContext } from 'react';

import { FeatureFlagsContext } from '../context/featureFlagsContext';

const useFeatureFlags = (...ids: string[]) => {
  const { enabledFlags } = useContext(FeatureFlagsContext);

  return ids.map((id) => enabledFlags.includes(id));
};

export default useFeatureFlags;
```

File: tsconfig.json [hunk 3]
```diff
@@ -22,5 +22,5 @@
     "types": ["@emotion/react/types/css-prop"]
   },
   "exclude": ["node_modules", ".next", "out"],
-  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx"]
+  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", "src/flags.js"]
```
Comment (mewdriller) [inline tsconfig.json:25]: ⚠️ Should this be written in TypeScript? And/or it looks like this doesn't exist at this point?

File: src/hooks/useFeatureToggle.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { useContext } from 'react';
+
+import { FeatureToggleContext } from '../context/featureToggleContext';
+
+const useFeatureToggle = () => {
```
Comment (mewdriller) [inline src/hooks/useFeatureToggle.ts:5]: 💭 I would expect a name like `useFeatureFlags` for this probably.

File: src/context/FeatureToggleProvider.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import { FC } from 'react';
+
+import { FeatureToggleContext } from '~/context/featureToggleContext';
+
+type FeatureToggleProps = {
+  children: React.ReactNode;
+  enabledFeatures: string[];
+};
+
+const FeatureToggle: FC<FeatureToggleProps> = ({
```
Comment (mewdriller) [inline src/context/FeatureToggleProvider.tsx:10]: 💭 I would expect a name like `FeatureFlagsProvider` for this.

File: src/context/featureToggleContext.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import { createContext } from 'react';
+
+export const FeatureToggleContext = createContext({
```
Comment (mewdriller) [inline src/context/featureToggleContext.ts:3]: 💭 I would expect a name like `FeatureFlagsContext` here.

File: src/pages/_app.tsx (hunk 1 above)
Comment (mewdriller) [reply src/pages/_app.tsx:39]: It's needed for things within the version of `@react-aria/*`  we're using to properly render in SSR. In more recent versions of React there is a `useId()` hook that removes the need for it.

**Review by mewdriller — COMMENTED** [review]
Here's an (outside) perspective on things. Let me know if you want to chat about the feedback at all.

File: src/pages/_app.tsx (hunk 1 above)
Comment (k8hansen) [reply src/pages/_app.tsx:39]: Thanks, we JUST updated to the new version of React but we were getting errors when Sean had another solution to our flab problem so we added it back in. 

Looks like we can remove it when we update `@react-aria` then? cc: @nelsW70

File: tsconfig.json (hunk 3 above)
Comment (k8hansen) [reply tsconfig.json:25]: Yes, thank you. There was an alleged reason this was just a JS file, but that was like 5 permutations ago, this is a leftover.

**Review by mewdriller — APPROVED** [review]
LGTM!

**Pull request merged**